src/measure_theory/function/strongly_measurable/supr.lean

```diff
@@ -0,0 +1,198 @@
+/-
+Copyright (c) 2023 Alex Kontorovich and Heather Macbeth. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Alex Kontorovich, Yury G. Kudryashov, Heather Macbeth
+-/
+import measure_theory.function.strongly_measurable.ae_sequence
+
+/-!
+# Supreme and infinite sums of (ae-)strongly measurable functions
+
+We prove lemmas for suprema and infima of `strongly_measurable` and `ae_strongly_measurable`
+functions, as well as, lemmas for their `tsum`s into `ennreal` and `nnreal`.
+
+-/
+
+open measure_theory filter topological_space function set measure_theory.measure
+open_locale ennreal topological_space measure_theory nnreal big_operators
+
+variables {α β ι : Type*} [measurable_space α] [topological_space β]
+
+section strongly_measurable
+
+open measure_theory set filter topological_space
+open_locale filter topological_space
+
+lemma finset.strongly_measurable_sup' {ι α β : Type*} [measurable_space α] [topological_space β]
+  [semilattice_sup β] [has_continuous_sup β] {f : ι → α → β} {s : finset ι} (hs : s.nonempty)
+  (hf : ∀ i ∈ s, strongly_measurable (f i)) : strongly_measurable (s.sup' hs f) :=
+finset.sup'_induction _ _ (λ _ h₁ _ h₂, h₁.sup h₂) hf
+
+lemma finset.strongly_measurable_sup'_pw {ι α β : Type*} [measurable_space α] [topological_space β]
+  [semilattice_sup β] [has_continuous_sup β] {f : ι → α → β} {s : finset ι} (hs : s.nonempty)
+  (hf : ∀ i ∈ s, strongly_measurable (f i)) : strongly_measurable (λ x, s.sup' hs (λ i, f i x)) :=
+by simpa only [← finset.sup'_apply] using finset.strongly_measurable_sup' hs hf
+
+/-- It would be nice to phrase this for `ι` of type `Prop` as well, but unfortunately this calls
+  `tendsto_finset_sup'_is_lub` which uses `finset ι`. -/
```

Comment:
This ought to be possible to avoid in the proof; perhaps by using `plift` somehow?

Reply:
I pushed a generalization; can you see if any other lemmas generalize well?